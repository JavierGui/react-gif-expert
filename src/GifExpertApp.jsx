import { useState } from "react";
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {

    const [Categories, setCategories] = useState([ 'One punch']);

    const onAddCategory = (newCategory) => {

       if(  Categories.includes( newCategory )  ) return;
        // Categories.push(newCategory);
          setCategories( [newCategory,...Categories]);
        // setCategories(cat => [...cat,'Valorant']);
    }

  return (
      <>
            
            <h1>GifExpertApp</h1>

            
            <AddCategory 
                    
                onNewCategory={ event => onAddCategory(event)}
                
            />

            
            
            
            { 
                Categories.map( (category) => (
                    <GifGrid
                    key={category} 
                    category={category}/>
                )) 
            }
                
            
                
      </>
      )
}
