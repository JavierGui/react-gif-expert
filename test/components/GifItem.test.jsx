import { render } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Pruebas en GifItem', () => { 

    const title = 'gojo';
    const url = 'https//jose-juan.com/saitama.jpg';

    test('Hacer match con el snapshot', () => { 

        const {container} = render(<GifItem title= {title} url= {url} />);
        expect( container ).toMatchSnapshot();

     });


     

 })