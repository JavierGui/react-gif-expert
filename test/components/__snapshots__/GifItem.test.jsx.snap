// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="gojo"
      src="https//jose-juan.com/saitama.jpg"
    />
    <p>
      gojo
    </p>
  </div>
</div>
`;
